python/lsst/meas/base/forcedPhotCoadd.py

```diff
@@ -399,16 +442,40 @@ def attachFootprints(self, sources, refCat, exposure, refWcs, dataRef):
 
         self.log.info("Loading deblended footprints for sources from %s, %s",
                       self.config.footprintDatasetName, dataRef.dataId)
-        fpCat = dataRef.get("%sCoadd_%s" % (self.config.coaddName, self.config.footprintDatasetName),
-                            immediate=True)
-        for refRecord, srcRecord in zip(refCat, sources):
-            fpRecord = fpCat.find(refRecord.getId())
-            if fpRecord is None:
-                raise LookupError("Cannot find Footprint for source %s; please check that %sCoadd_%s "
-                                  "IDs are compatible with reference source IDs" %
-                                  (srcRecord.getId(), self.config.coaddName,
-                                   self.config.footprintDatasetName))
-            srcRecord.setFootprint(fpRecord.getFootprint())
+
+        if self.config.footprintDatasetName == "scarletDeblendModels":
+            # Load the scarlet models
+            dataModel = dataRef.get("%sCoadd_%s" % (self.config.coaddName, self.config.footprintDatasetName),
+                                    immediate=True)
+            self._attachScarletFootprints(refCat, dataModel, exposure, dataRef.dataId["band"])
+        else:
+            fpCat = dataRef.get("%sCoadd_%s" % (self.config.coaddName, self.config.footprintDatasetName),
+                                immediate=True)
+            for refRecord, srcRecord in zip(refCat, sources):
+                fpRecord = fpCat.find(refRecord.getId())
+                if fpRecord is None:
+                    raise LookupError("Cannot find Footprint for source %s; please check that %sCoadd_%s "
+                                      "IDs are compatible with reference source IDs" %
+                                      (srcRecord.getId(), self.config.coaddName,
+                                       self.config.footprintDatasetName))
+                srcRecord.setFootprint(fpRecord.getFootprint())
```

Comment:
You had a comment above that said that `attachFootprints` was Gen2-specific and wasn't worth modifying (which sounds plausible, but isn't something I can confirm or deny off the top of my head).  Did that turn out not to be true?

Reply:
That wasn't my comment, but one that has been in the repo since Nate updated this task 3 years ago. I think his point is that this method loads the footprint file from the data ref, which is not the way that gen3 works. So I think he was saying that when we officially deprecate gen2 we can remove this method, but in the meantime I wanted it to work with the scarlet outputs for consistency.